internal/controller/ai_gateway_route.go

```diff
@@ -437,47 +435,41 @@ func (c *AIGatewayRouteController) newHTTPRoute(ctx context.Context, dst *gwapiv
 		},
 	}}
 	var rules []gwapiv1.HTTPRouteRule
-	for _, rule := range aiGatewayRoute.Spec.Rules {
+	for i, rule := range aiGatewayRoute.Spec.Rules {
+		routeName := routeName(aiGatewayRoute, i)
+		var backendRefs []gwapiv1.HTTPBackendRef
+		var timeouts *gwapiv1.HTTPRouteTimeouts
 		for i := range rule.BackendRefs {
 			br := &rule.BackendRefs[i]
 			dstName := fmt.Sprintf("%s.%s", br.Name, aiGatewayRoute.Namespace)
-			if _, ok := dedup[dstName]; ok {
-				continue
-			}
-			dedup[dstName] = struct{}{}
-
-			var backendRefs []gwapiv1.HTTPBackendRef
-			var timeouts *gwapiv1.HTTPRouteTimeouts
 			if isInferencePoolRef(br) {
-				// When the target is InferencePool, we don't need the HTTPRoute level setting, but
-				// will route to ORIGINAL_DST, so we don't need to set the backendRefs.
-				//
-				// However, to properly route to the ORIGINAL_DST, we use the constant OriginalDstClusterName
-				// as the selected backend name.
-				dstName = extensionserver.OriginalDstClusterName
-
-				// TODO: make the timeout configurable. One way is to move the timeout setting from
-				// 	AIServiceBackend to AIGatewayRouteBackendRef.
-				timeouts = inferencePoolDefaultTimeout
+				c.logger.Info("TODO: inference pool ref not supported yet",
+					"namespace", aiGatewayRoute.Namespace, "name", br.Name)
 			} else {
 				backend, err := c.backend(ctx, aiGatewayRoute.Namespace, br.Name)
 				if err != nil {
 					return fmt.Errorf("AIServiceBackend %s not found", dstName)
 				}
-				backendRefs = []gwapiv1.HTTPBackendRef{
-					{BackendRef: gwapiv1.BackendRef{BackendObjectReference: backend.Spec.BackendRef}},
-				}
-				timeouts = backend.Spec.Timeouts
+				backendRefs = append(backendRefs,
+					gwapiv1.HTTPBackendRef{BackendRef: gwapiv1.BackendRef{
+						BackendObjectReference: backend.Spec.BackendRef,
+						// TODO: change the weight's type to int32 in the API.
+						Weight: ptr.To[int32](int32(br.Weight)), // nolint:gosec
+					}},
+				)
+				// For now, if there're multiple backends with different timeouts, we just take the first one.
+				// TODO: deprecate the AIServiceBackend timeout and move it to Rule level.
```

Comment:
Do you mean the AIGatewayRoute.rules?

Reply:
https://github.com/envoyproxy/ai-gateway/pull/599/commits/372b671331c8fd2064b81cbdbbb225660b4d98aa clarified